packages/quantic/force-app/main/default/lwc/quanticTabBar/quanticTabBar.js

```diff
@@ -0,0 +1,361 @@
+      this.hideMoreButton();
+      this.hasRendered = true;
+    }
+  }
+
+  /**
+   * Updates the display of the tabs.
+   * @returns {void}
+   */
+  updateTabsDisplay = () => {
+    if (this.isOverflow) {
+      this.showMoreButton();
+    } else {
+      this.hideMoreButton();
+    }
+    this.updateTabVisibility(this.overflowingTabs, false);
+    this.updateTabVisibility(this.displayedTabs, true);
+    this.updateComboboxOptions();
+    this.updateMoreButtonPosition();
+    this.isComboboxOpen = false;
+  };
+
+  /**
+   * Updates the combobox options.
+   * @returns {void}
+   */
+  updateComboboxOptions() {
+    this.options = this.overflowingTabs.map((el) => ({
+      // @ts-ignore
+      label: el.label,
+      // @ts-ignore
+      value: el.expression,
+    }));
+  }
+
+  /**
+   * Updates the position of the more button element.
+   * @returns {void}
+   */
+  updateMoreButtonPosition() {
+    this.moreButton?.style.setProperty(
+      'left',
+      `${this.displayedTabs.length ? this.lastVisibleTabRightPosition : 0}px`
+    );
+  }
+
+  /**
+   * Updates the state of the more button element.
+   * @returns {void}
+   */
+  updateMoreButtonState() {
+    if (this.hasRendered) {
+      if (this.maxMoreButtonWidth < this.moreButtonWidth) {
+        this.maxMoreButtonWidth = this.moreButtonWidth;
+      }
+
+      this.expandedMoreButton =
+        this.containerWidth > this.maxMoreButtonWidth + this.selectedTabWidth;
+    }
+  }
+
+  /**
+   * Shows the more button.
+   * @returns {void}
+   */
+  showMoreButton() {
+    this.moreButton?.style.setProperty('display', 'block');
+  }
+
+  /**
+   * Hides the more button.
+   * @returns {void}
+   */
+  hideMoreButton() {
+    this.moreButton?.style.setProperty('display', 'none');
+  }
+
+  /**
+   * Updates the tabs visibility.
+   * @param {Array<Element>} tabElements
+   * @param {boolean} isVisible
+   */
+  updateTabVisibility(tabElements, isVisible) {
+    tabElements.forEach((tab, index) => {
+      const tabsCount = this.getTabsFromSlot().length;
+      // @ts-ignore
+      tab.style.setProperty(
+        'order',
+        isVisible ? index + 1 : tabsCount - tabElements.length + index + 1
+      );
+      // @ts-ignore
+      tab.style.setProperty('visibility', isVisible ? 'visible' : 'hidden');
+    });
+  }
+
+  /**
+   * Indicates whether the tabs are causing an overflow.
+   * @returns {boolean}
+   */
+  get isOverflow() {
+    return this.slotContentWidth >= this.containerWidth;
+  }
+
+  /**
+   * Returns the tab bar container element.
+   * @returns {Element}
+   */
+  get container() {
+    return this.template.querySelector('.tab-bar_container');
+  }
+
+  /**
+   * Returns the container's width.
+   * @returns {number}
+   */
+  get containerWidth() {
+    return this.getAbsoluteWidth(this.container);
+  }
+
+  /**
+   * returns the width of the content of the slot.
+   * @returns {number}
+   */
+  get slotContentWidth() {
+    return this.getTabsFromSlot()?.reduce(
```

Comment:
My understanding is that `getTabsFromSlot` always return an array.

```suggestion
    return this.getTabsFromSlot().reduce(
```